css/tokens/index.js

```diff
@@ -34,39 +28,127 @@ async function createTokens() {
  */
-function collectTokenSources(rendered) {
-	const collection = rendered.stats.includedFiles
-		.map((/** @type {string} */ file) => {
-			return { name: getFinalFilePart(file), location: getAtlasFilePath(file) };
-		})
-		.filter((/** @type {{ name: string; location: string; }} */ x) => x.name !== 'index')
-		.reduce(
+function getSortedOrder(collection) {
+	return Object.fromEntries(Object.entries(collection).sort());
+}
+
+/**
+ *
+ * @param {string[]} paths token file paths
+ */
+// Print a warning if a token file lacks the required sass-export-section comments.
+function checkFileComments(paths) {
+	paths.forEach(path => {
+		try {
+			fs.readFile(path, 'utf8', function read(err, result) {
+				if (err) {
+					throw err;
+				}
+				if (!result.includes('@sass-export-section')) {
+					console.log(`Warning: ${path} is missing @sass-export-section annotations.`);
+				}
+			});
+		} catch (err) {
+			throw new Error(`Problem reading token files: ${err}`);
+		}
+	});
+}
+
+/**
+ *
+ * @param {import('./types').SassExportTokens} tokens raw token data by group
+ * @returns {import('./types').SassExportCollection}
+ */
+function collectTokens(tokens) {
+	/** @type {import('./types').SassExportCollection} */
+	const collection = {};
+	for (const [parent, tokenValues] of Object.entries(tokens)) {
+		//Currently using sass-export-section annotations in the token files for grouping.
+		//Tokens without annotations will be combined in the variables array.
+		if (parent === 'variables') continue;
+
+		const collectedValues = tokenValues.reduce(
 			(
-				/** @type {{ [x: string]: any; }} */ all,
-				/** @type {{ name: string | number; }} */ source
+				/** @type {import('./types').SassExportCollection } */ all,
+				/** @type {import('./types').SassExportTokenItem} */ current
 			) => {
-				all[source.name] = {
-					...source,
-					tokens: {}
+				const tokenName = current.name;
+
+				/** @type {string | import('./types').SassExportTokenNestedItem} */
+				const values = current.mapValue
+					? { ...getNestedTokens(current) }[tokenName]
+					: current.compiledValue;
+
+				all[parent] = {
+					...all[parent],
+					[tokenName]: values
 				};
 				return all;
 			},
 			{}
 		);
+		collection[parent] = {
+			name: parent,
+			location: `/css/src/tokens/${parent}.scss`,
+			tokens: collectedValues[parent]
+		};
+	}
 	return collection;
 }
 
+/**
+ *
+ * @param {import('./types').SassExportTokenItem} child
+ * @returns {{[key: string]: import('./types').SassExportTokenNestedItem}}
+ */
+function getNestedTokens(child) {
+	if (!child.mapValue) {
+		const { name, compiledValue, value } = child;
+		return { [name]: compiledValue ? compiledValue : value };
+	}
+	/** @type {{[key: string]: import('./types').SassExportTokenNestedItem}} */
+	const childMap = {};
+	child.mapValue.forEach(subChild => {
+		childMap[child.name] = { ...childMap[child.name], ...getNestedTokens(subChild) };
+	});
+	return childMap;
+}
+
```

Comment:
Unfortunately, after looking at themes, this update will need some fairly major changes. I think we need the values here, rather only the variable names.

![image](https://user-images.githubusercontent.com/30843002/128909677-4bd75f11-697b-4d2f-b7f9-a7c0efc28830.png)

Reply:
Add a section to parse compiledValue and retrieve the values.  
![image](https://user-images.githubusercontent.com/64044661/128952629-7ec19fa9-f569-4262-a27e-c7ab7ba15475.png)

Reply:
Also coerced booleans 
![image](https://user-images.githubusercontent.com/64044661/128952734-f2a7202c-035e-465d-a8a9-ed71f6adf81c.png)